src/components/dev-hub/star-rating.js

```diff
@@ -0,0 +1,180 @@
+    margin-right: ${size.mediumLarge};
+    box-sizing: content-box;
+
+    @media ${screenSize.mediumAndUp} {
+        padding: 4px;
+        width: auto;
+        height: auto;
+        margin-right: ${({ isActive }) => (isActive ? '0' : size.xsmall)};
+
+        &:hover {
+            cursor: ${({ isClicked }) => (isClicked ? 'default' : 'pointer')};
+        }
+    }
+`;
+
+const StyledStarsContainer = styled.div`
+    display: flex;
+    align-items: center;
+
+    @media ${screenSize.mediumAndUp} {
+        margin-left: ${size.large};
+    }
+`;
+
+const StyledTitle = styled.span`
+    font-family: 'Fira Mono', monospace;
+    font-weight: bold;
+    font-size: ${fontSize.small};
+`;
+
+const initialState = {
+    first: false,
+    second: false,
+    third: false,
+    fourth: false,
+    fifth: false,
+    isClicked: false,
+};
+
+const reducer = (state, action) => {
+    let nextState = { ...initialState };
+    switch (action) {
+        case 'fifth':
+            nextState = { ...nextState, fifth: true };
+        case 'fourth':
+            nextState = { ...nextState, fourth: true };
+        case 'third':
+            nextState = { ...nextState, third: true };
+        case 'second':
+            nextState = { ...nextState, second: true };
+        case 'first':
+            nextState = { ...nextState, first: true };
+            break;
+        case 'clicked':
+            nextState = { ...state, isClicked: true };
+            break;
+        case 'clear':
+            nextState = { ...initialState };
+            break;
+        default:
+            throw new Error();
+    }
+
+    return nextState;
+};
+
+const StarRating = ({ onFirstsClick, onMiddleClick, onLatestClick }) => {
+    const [state, dispatch] = useReducer(reducer, initialState);
+
+    const onClickHandler = useCallback(
+        handler => {
+            if (!state.isClicked) {
+                handler && handler();
+                dispatch('clicked');
+            }
+        },
+        [state.isClicked]
+    );
+
+    const onHoverHandler = useCallback(
+        star => {
+            !state.isClicked && dispatch(star);
+        },
+        [state.isClicked]
+    );
+    const onLeaveHandler = useCallback(() => {
+        !state.isClicked && dispatch('clear');
+    }, [state.isClicked]);
+
+    return (
+        <StyledContainer>
+            <StyledTitle>Rate this article</StyledTitle>
+            <StyledStarsContainer onMouseLeave={onLeaveHandler}>
+                <StyledStar
+                    isActive={state.first}
+                    isClicked={state.isClicked}
+                    name="first-star"
+                    onClick={() => onClickHandler(onFirstsClick)}
+                    onMouseEnter={() => onHoverHandler('first')}
+                    width={state.first && size.mediumLarge}
+                >
+                    <stop offset="0%" stopColor="#D34F94" />
+                    <stop offset="100%" stopColor="#D95B8F" />
```

Comment:
Perhaps we could re-use these in a new component above and pass the start and end colors as props for the gradient? (Or even better, we could turn one full star into a reusable component)

Reply:
I would like to keep this variant. If we need some manipulation with colors this variant allow change   amount of colors and set offset for them.